src/coreclr/nativeaot/Runtime/unix/HardwareExceptions.cpp

```diff
@@ -598,26 +613,30 @@ bool InitializeHardwareExceptionHandling()
     {
         return false;
     }
+#ifdef TARGET_UNIX
+    Thread* curThread = ThreadStore::RawGetCurrentThread();
+    curThread->EnsureSignalAlternateStack();
```

Comment:
`InitializeHardwareExceptionHandling` runs once per process. It does not run for each thread.

Try to trigger the stackoverflow on a secondary thread. I do not think it is going to be reported correctly.

Reply:
There's a call to EnsureSignalAlternateStack in the Thread::Construct method that should handle new threads. It reports Stack Overflow in a repro project that overflows in a new thread.

Reply:
Ah ok. Why do we need to call it here? Should the call in Thread::Construct be enough?

Reply:
Good point, it looks like the Thread::Construct call is enough.